lib/utils.js

```diff
@@ -163,4 +166,67 @@ function getAlias(loader, canonicalName) {
   });
 
   return bestAlias || canonicalName;
-}
\ No newline at end of file
+}
+
+function getGraphEntryPoints(graph, entryPoints) {
+  entryPoints = [].concat(entryPoints || []);
+
+  var modules = Object.keys(graph.adjList);
+  var discarded = {};
+
+  modules.forEach(function (moduleName) {
+    Object.keys(graph.adjList[moduleName]).forEach(function (depName) {
+      discarded[depName] = true;
+    });
+  });
+
+  modules.filter(function (moduleName) {
+    return !discarded[moduleName];
+  }).sort().forEach(function (moduleName) {
+    if (entryPoints.indexOf(moduleName) === -1) {
+      entryPoints.push(moduleName);
+    }
+  });
+
+  return entryPoints;
+}
+
+exports.postOrderTraverseTree = function postOrderTraverseTree(tree, entryPoints) {
```

Comment:
Perhaps call this `getTreeModulesPostOrder` or something like that? Strictly we're not really traversing, but ordering?